### DISCUSSION TITLE
Four Limit Switches and the Error Still Arises...

### DISCUSSION THREAD

**Discussion by silver2row** [grblHAL/core#221] [General]
Hello,

I added four limit switches to the board, i.e. NC accessible switches. I am still getting the first starter error.

I know what the docs. say. I have read them over and over. I just added my 12v PSU, some motors, and four limit switches.

If I solder on the screw terminals, do each of the soldered on screw terminals need an I/O or can I just leave them floating?

Seth

P.S. I have started a discussion before in here but thought this would solve the issue.

**Comment by terjeio** [comment]
The "first starter" error is due to the reset (or estop) input beeing asserted, not the limit switches. Connect it to ground or invert it.

> If I solder on the screw terminals, do each of the soldered on screw terminals need an I/O or can I just leave them floating?

I am not sure I understand this... If a terminal is not connected to an I/O pin it does not matter if it is connected or not. If a terminal connected to an input pin it should be wired to ground or high level voltage (3.3V) if directly connected to a MCU pin as the pull-up/pull-down resistors are rather large in value (typically ~50K). The best is to have optocouplers between the terminal and the MCU pin.

**Reply by silver2row** [reply]
Hello,

Sorry for my late reply. What should I connect to GND and/or what should I invert? I currently have three motors and four endstops.

I read the article on how to attach four endstops to handle this error. I have not yet made it past this error.

Seth

P.S. I am using the board and a Teensy4.1 attached to the board directly from the headers.

**Reply by terjeio** [reply]
Did you try the estop/reset/halt input (I do not know what it is labeled on your board).
Do you get an error or alarm, and which number?

**Reply by silver2row** [reply]
Hello,

I have not tried anything yet. I just added in the motors and endstops, PSU. I am using the GRBLhal Board from Tindie w/ the Teensy4.1 attachment. 

I will look for the estop/halt input. Thank you. 

Seth

**Reply by silver2row** [reply]
I will update you soon. Sorry it has taken me so long to let you know what is going on.

Seth
